### ISSUE TITLE
Download crashes when trying to download all images: "isic image download images/ -l 0"

### ISSUE THREAD

**Issue by PascalSchimmler** [ImageMarkup/isic-cli#4]
I can download smaller amountsm, but it alwasys crashes when downloading all images, this time with 51 seconds to go.

```
File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\requests\models.py", line 760, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\urllib3\response.py", line 579, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\urllib3\response.py", line 544, in read
    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
  File "C:\Users\x\AppData\Local\Programs\Python\Python39\lib\contextlib.py", line 137, in __exit__
    self.gen.throw(typ, value, traceback)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\urllib3\response.py", line 446, in _error_catcher
    raise ReadTimeoutError(self._pool, None, "Read timed out.")
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='content.isic-archive.com', port=443): Read timed out.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\x\pycharmprojects\isic-cli\isic_cli\cli\__init__.py", line 132, in main
    cli()
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\Pascal\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\click\decorators.py", line 38, in new_func
    return f(get_current_context().obj, *args, **kwargs)
  File "c:\users\x\pycharmprojects\isic-cli\isic_cli\cli\utils.py", line 17, in decorator
    f(ctx, **kwargs)
  File "c:\users\x\pycharmprojects\isic-cli\isic_cli\cli\image.py", line 99, in download
    Parallel()(delayed(download_image)(image, outdir, progress, task2) for image in images)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\joblib\parallel.py", line 1056, in __call__
    self.retrieve()
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\joblib\parallel.py", line 935, in retrieve
    self._output.extend(job.get(timeout=self.timeout))
  File "C:\Users\x\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 771, in get
    raise self._value
  File "C:\Users\x\AppData\Local\Programs\Python\Python39\lib\multiprocessing\pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\joblib\_parallel_backends.py", line 595, in __call__
    return self.func(*args, **kwargs)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\joblib\parallel.py", line 262, in __call__
    return [func(*args, **kwargs)
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\joblib\parallel.py", line 262, in <listcomp>
    return [func(*args, **kwargs)
  File "c:\users\pascal\pycharmprojects\isic-cli\isic_cli\io\http.py", line 112, in download_image
    for chunk in r.iter_content(1024 * 1024 * 5):
  File "C:\Users\x\PycharmProjects\HCI_Home5\isic-cli\lib\site-packages\requests\models.py", line 767, in generate
    raise ConnectionError(e)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='content.isic-archive.com', port=443): Read timed out.

isic-cli: v2.0.1.dev1+g2dd86f9
python:   v3.9.12
time:     2022-06-05T09:15:26.758860
os:       Windows-10-10.0.19043-SP0
env:      -
user:     -
command:  isic image download images3/ -l 0
```

**Comment by danlamanna** [comment]
Sorry for this, when making tens of thousands of requests there tend to be a few failures. We perform automatic retries in a number of cases but it appears we missed one.

A new version is out with significantly better retry logic. You can `pip install --upgrade isic-cli` to get it. Let me know if it works for you.

In the next couple of days I'll push a new release that also allows a download command to pick up where it left off, which should also help if this occurs again.

**Comment by danlamanna** [comment]
> In the next couple of days I'll push a new release that also allows a download command to pick up where it left off, which should also help if this occurs again.

This feature should be available now in v3.0.0.

**Comment by danlamanna** [comment]
> Thank you for the fast and extensive reply, I already found another repo that can download the dataset, but I'm gonna give this a try

Which repo was it if you don't mind me asking? Most of these projects use the old ISIC API which is being decommissioned soon.
